### PR TITLE
refact(skymp5-server): minor fixes, refactoring of PapyrusVM

### PR DESCRIPTION
**Pull request by Darkwood9612**

### CODE REVIEW

File: skymp5-server/cpp/papyrus_vm_lib/VarValue.cpp [hunk 1]
```diff
@@ -43,141 +40,108 @@ VarValue VarValue::CastToFloat() const
...
 {
   if (!promise) {
     throw std::runtime_error("Not a promise");
   }
   promise->Then(cb);
 }
 
-VarValue::VarValue(uint8_t type)
+namespace  {
+     constexpr inline std::string_view EMPTY_STR = "";
```
Comment (Pospelove) [inline skymp5-server/cpp/papyrus_vm_lib/VarValue.cpp:77]: EMPTY_STR => kEmptyString

**Review by Pospelove — CHANGES_REQUESTED** [review]
Overall looks good!

File: unit/VarValueTest.cpp [hunk 2]
```diff
@@ -227,7 +227,7 @@ TEST_CASE("Cast to string", "[VarValue]")
   REQUIRE(CastToString(VarValue(5.0)) == VarValue("5"));
   REQUIRE(CastToString(VarValue(4278190080.0)) == VarValue("4278190080"));
 
-  VarValue arr((uint8_t)VarValue::kType_ObjectArray);
+  VarValue arr((uint8_t)VarValue::Type::ObjectArray);
```
Comment (nic11) [inline unit/VarValueTest.cpp:230]: C-style cast

**Comment by Pospelove** [comment]
Я не могу понять что но что то серьёзное развалилось (

**Comment by Pospelove** [comment]
Sorry :(
I wasn't able to determine the reason for failure. We will probably get back to this PR once we have better tooling to detect the reason. Closing for now :((

**Pull request closed without merging**